src/store/api/github.js

```diff
@@ -0,0 +1,78 @@
+import { Octokit } from 'octokit';
+import { createApi, fakeBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export const octokit = new Octokit({
+  auth: import.meta.env.VITE_GITHUB_API_TOKEN
+});
+
+/**
+ * Return all commits from a Github repository
+ * identified by owner and name.
+ */
+async function listCommits(owner, repo) {
+  const res = await octokit.rest.repos.listCommits({
+    owner, repo, per_page: 4,
+  });
+
+  return res.data.map(({ commit }) => ({
+    message: commit.message,
+    author: commit.author.name ?? commit.author.login,
+    date: new Date(commit.committer.date),
+    repo, owner
+  }));
+}
+
+/**
+ * The Github api slice
+ *
+ * It does not use a base query, since the octokit API
+ * does not allow for a nice abstraction in endpoint form
+ */
+const github = createApi({
+  reducerPath: 'github',
+  baseQuery: fakeBaseQuery(),
+  endpoints: build => ({
+    allCommits: build.query({
+      queryFn: async () => {
```

Comment:
Maybe you want to extract this allCommits logic in its own function?  The same hold for the other endpoints. This will become big with new endpoints